eth-contracts/contracts/Governance.sol

```diff
@@ -332,44 +341,41 @@ contract Governance is RegistryContract {
         return proposals[_proposalId].votes[_voter];
     }
 
-    // ========================================= Private =========================================
+    // ========================================= Internal =========================================
 
-    function _executeTransaction(
-        address _targetContractAddress,
-        uint256 _callValue,
-        string memory _signature,
-        bytes memory _callData
-    ) internal returns (bytes memory /** returnData */)
+    function _executeTransaction(uint256 _proposalId) internal
+    returns (bool /** success */)
     {
+        address targetContractAddress = proposals[_proposalId].targetContractAddress;
+        uint256 callValue = proposals[_proposalId].callValue;
+        string memory signature = proposals[_proposalId].signature;
+        bytes memory callData = proposals[_proposalId].callData;
+
         bytes32 txHash = keccak256(
             abi.encode(
-                _targetContractAddress, _callValue, _signature, _callData
+                targetContractAddress, callValue, signature, callData
             )
         );
 
-        bytes memory callData;
-
-        if (bytes(_signature).length == 0) {
-            callData = _callData;
+        bytes memory encodedCallData;
+        if (bytes(signature).length == 0) {
+            encodedCallData = callData;
         } else {
-            callData = abi.encodePacked(bytes4(keccak256(bytes(_signature))), _callData);
+            encodedCallData = abi.encodePacked(bytes4(keccak256(bytes(signature))), callData);
         }
 
         (bool success, bytes memory returnData) = (
             // solium-disable-next-line security/no-call-value
-            _targetContractAddress.call.value(_callValue)(callData)
+            targetContractAddress.call.value(callValue)(encodedCallData)
```

Comment:
how does this handle the issue of 'proxyDeployerAdmin' being the only allowed caller to 'upgradeTo'? Looks like the same exact code as before with bool return, arguments removed and accessed from the proposal struct instead...